src/lib/query.ts

```diff
@@ -118,111 +118,256 @@ public static async queryRecordTypes(conn: Connection): Promise<String[]> {
     })
     }
 
-public static async queryStdPricebookEntryIds(conn: Connection, productList: Set<String>): Promise<String[]> {
-        Util.log('--- exporting standard Pricebook Entry Ids');
-        if(productList.size >90){
-            let paramsObject1: Query={     
-                "queryBegining": "SELECT Id FROM PricebookEntry WHERE Product2.Name IN (",
-                "queryConditions": ") AND Pricebook2Id != null AND Pricebook2.IsStandard = true",
-                "objectsList": productList,
-                "sobjectName": "standard Pricebook Entry Ids"
-            }
-            let paramsObject2: Query={     
-                "queryBegining": "SELECT Id FROM PricebookEntry WHERE Product2.enxCPQ__Root_Product__r.Name IN (",
-                "queryConditions": ") AND Pricebook2Id != null AND Pricebook2.IsStandard = true",
-                "objectsList": productList,
-                "sobjectName": "standard Pricebook Entry Ids"
-            }
-            return await Util.createQueryPromiseArray(paramsObject1, conn, paramsObject2);
+    public static async queryTargetProductIds(conn: Connection,  techIds: Set<String>): Promise<String[]> {
+        Util.log('--- exporting target productIds');
+        if(techIds.size>90){
+    
+            let paramsObject1: Query ={     
+              "queryBegining": "SELECT enxCPQ__TECH_External_Id__c FROM Product2 WHERE enxCPQ__TECH_External_Id__c IN (",
+              "queryConditions": ") AND RecordType.Name IN ('Product', 'Bundle', 'Option','Resource')",
+              "objectsList": techIds,
+              "sobjectName": "target productIds",
+          }
+  
+            return await Util.createQueryPromiseArray(paramsObject1, conn);
+        }
+    
+        let query = "SELECT enxCPQ__TECH_External_Id__c FROM Product2 WHERE enxCPQ__TECH_External_Id__c IN (" + Util.setToIdString(techIds) + ") AND RecordType.Name IN ('Product', 'Bundle', 'Option','Resource')";
+            return new Promise<String[]>((resolve: Function, reject: Function) => {
+            conn.query(query, 
+                null,
+                function (err, res) {
```

Comment:
Just curiosity - why here you're using "function(params)" notation and couple lines below you're writing arrow function?